pkg/operator/ceph/object/bucket/provisioner.go

```diff
@@ -385,7 +385,8 @@ func (p *Provisioner) composeObjectBucket() *bktv1alpha1.ObjectBucket {
 
 	return &bktv1alpha1.ObjectBucket{
 		Spec: bktv1alpha1.ObjectBucketSpec{
-			Connection: conn,
+			StorageClassName: storageClassName,
```

Comment:
The lib bucket provisioner sets storageclass name in https://github.com/kube-object-storage/lib-bucket-provisioner/blob/master/pkg/provisioner/controller.go#L383 after `Provisioner` or `Grant`. So this change may be reductant

Reply:
That is only set **after** the Secret and ConfigMap are created, so if those or anything before it fails, the cleanup function will have empty information and fail to clean up the resource properly.

Reply:
From your description sounds like a good thing to update the lib-bucket-provisioner, but I haven't looked in more detail.